### DISCUSSION TITLE
Sharing your idea to get an intuition

### DISCUSSION THREAD

**Discussion by beginner984** [milaboratory/mixcr#854] [Ideas]
Hello mixcr team

I read that `10x genomics cellranger returns some output from count for GEX (5' and 3') and VDJ for enriched immune receptor libraries` which could be visualised by tools like Platypus, etc

So, having this, what is the purpose of tools like mixcr please? I mean, when 10x genomics cellranger gives these things why one should go to mixcr like software

Am I right mixcr and cellranger can do the same tasks?

Sorry I am very naive in immunology

Thanks for any help in advance

**Comment by PoslavskySV** [comment]
Hi,

there are multiple reasons. Among them:

 - single cell is not limited to 10x, there are many of protocols and technologies, which MiXCR either already supports or is flexile enough to work with such data; cell ranger obviously is limited to 10x only;
 - algorithms for VDJ repertoire reconstruction are far more advanced in MiXCR than in cell ranger or any other software; they are tested through years on millions and millions of VDJ datasets obtained with variety of wet lab protocols and sequencing technologies; MiXCR is aware of all possible wet lab peculiarities for different protocols;
 - there are many dedicated features for repertoire profiling in MiXCR that simply absent in other tools;
 - and the functionality for GEX and phenotyping is coming soon in MiXCR.

Other people may add different reasons.

**Comment by mizraelson** [comment]
Well, briefly cellranger is originally a tool for a single cell gene expression analysis which works with T/B cell receptors as an additional feature. MIXCR is a powerful software specifically designed for immune repertoire analysis.
Because TCR and BCR repertoires are so diverse the analysis requires a very high precision to separate various errors (sequencing, PCR etc. ) from true VDJ variants which is not the case in a regular gene expression data. MiXCR  incorporates powerful aligning, filtering and assembling algorithms (not to mention our own genes' segments databases for multiple species) that  outperform other tools in back-to-back-testing and eliminate all the false positive clones. 
MiXCR also includes an advanced downstream repertoire analysis  again specifically designed for the task. 
Since most of us, who develop the software, have a huge immunogenetics background in academia we have a very deep understanding what features, types of data normalization and statistical analysis are required, and do our best to deliver the whole complexity of it in an easy to use interface. All the features we have - we develop from scratch for the best efficiency, quality and performance which is impossible to achieve by combining a number of 3rd party tools.

**Comment by github-wow** [comment]
Hopefully, it would be appropriate to ask here. MIXCR has a lot of parameters. For example: [downsampling](https://docs.milaboratories.com/mixcr/reference/mixcr-postanalysis/#downsampling) can be done at 3 levels: `count`, `top`, `cumtop`. Each of the parameter can have 3 options: `read`, `umi` or `cell`. And `count` has additional suboptions: `auto`, `min` and `fixed`. What is the intuitive way to understand what parameters is necessary and sufficient ?
